awx_collection/plugins/modules/tower_user.py

```diff
@@ -128,7 +127,7 @@ def main():
         email=dict(required=False, type='str'),
         is_superuser=dict(required=False, type='bool', default=False, aliases=['superuser']),
         is_system_auditor=dict(required=False, type='bool', default=False, aliases=['auditor']),
-        password=dict(required=False, type='str', default=''),
```

Comment:
Why do we not want the default of '' to bleed through from the Tower API into the modules?

Reply:
I believe we should destroy all defaults. I strain to think of a single case where any of the parameters should have a default. They are an anti-feature, a burden.

If the API has a default, then that categorically means that the module does not need to provide it. If the module adds such a default, then it exposes itself to the risk of unwanted resetting of fields on modification. It makes the modules far less predictable than not including the keys.

I will concede this possibility for defaults - `organization` fields could have a default of "Default". This is because Tower installs have this data pre-loaded. Same could be said for other resource types (maybe). Historically, some license types only _had_ 1 organization, so requiring the user to provide the field is incoherent to such users.

That brings me back to required-ness. If a field is required, it shouldn't have a default. That's over-specified, it doesn't make sense.

What fields should be required? Only the identity fields. The identity fields are only knowable from the Named URL mapping, due to customizations in this app. Those will be required for lookup, thus, they must be provided every time. If it's not a lookup field, then there is always a scenario (of modification) where it should be omitted, thus, it shouldn't have a default.

There is another case - `playbook`, a field required on creation but not on modification. That leads to a more nuanced criteria of "required". Still, you don't want a default.